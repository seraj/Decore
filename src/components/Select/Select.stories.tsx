import { boolean, withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";
import wInfo from "../../utils/wInfo";
import Select from "./Select";

const stories = storiesOf("Components/Select", module);
stories.addDecorator(withKnobs);

stories.add(
  "default",
  wInfo()(() => (
    <Select
      disabled={boolean("disabled", false)}
      success={boolean("success", false)}
      error={boolean("error", false)}
      fluid={boolean("fluid", false)}
    >
      <option value="test" selected>
        selected
      </option>
      <option value="disable" disabled>
        disabled
      </option>
      <option value="normal">Normal</option>
    </Select>
  ))
);
