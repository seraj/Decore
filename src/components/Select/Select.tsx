import * as React from "react";
import StyledSelect from "./styled/StyledSelect";
import SelectProps from "./Select.props";

const Select: React.FC<SelectProps> = ({ children, ...props }) => (
  <StyledSelect {...props}>{children}</StyledSelect>
);

export default Select;
