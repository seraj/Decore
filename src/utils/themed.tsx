import * as React from "react";
import * as renderer from "react-test-renderer";
import Decore from "../components/Decore";
import theme from "../theme";

const themed = (component: JSX.Element) =>
  renderer.create(<Decore theme={theme}>{component}</Decore>);

export default themed;
