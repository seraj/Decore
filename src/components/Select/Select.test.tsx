import * as React from "react";

import themed from "../../utils/themed";
import Select from "./Select";

describe("Select", () => {
  it("should render correctly", () => {
    const component = themed(<Select>test</Select>).toJSON();

    expect(component).toMatchSnapshot();
  });
});
