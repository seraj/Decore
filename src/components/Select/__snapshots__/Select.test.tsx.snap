// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render correctly 1`] = `
.c0 {
  color: #333333;
  padding: .6em 2.2em .5em .8em;
  height: 45px;
  border: 1.5px solid #999999;
  border-radius: 0px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: undefined;
  background-image: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTgiIGhlaWdodD0iMTEiIHZpZXdCb3g9IjAgMCAxOCAxMSIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj48cGF0aCBkPSJNMSAxTDkgOS42MTA5MUwxNyAxIiBzdHJva2U9IiM5OTk5OTkiIHN0cm9rZS13aWR0aD0iMS41IiBzdHJva2UtbWl0ZXJsaW1pdD0iMTAiIHN0cm9rZS1saW5lam9pbj0iYmV2ZWwiLz48L3N2Zz4=);
  background-repeat: no-repeat,repeat;
  background-position: right .7em top 50%,0 0;
  background-size: .9em auto,100%;
}

.c0:hover {
  background-image: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGZpbGw9Im5vbmUiIHZpZXdCb3g9IjAgMCAxOCAxMSI+ICAgIDxwYXRoIHN0cm9rZT0iIzMzMyIgc3Ryb2tlLWxpbmVqb2luPSJiZXZlbCIgc3Ryb2tlLW1pdGVybGltaXQ9IjEwIiBzdHJva2Utd2lkdGg9IjEuNSIgZD0iTTEgMUw5IDkuNjEwOTFMMTcgMSIvPjwvc3ZnPg==);
  border-color: #333333;
}

.c0:disabled {
  background-image: url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGZpbGw9Im5vbmUiIHZpZXdCb3g9IjAgMCAxOCAxMSI+ICAgIDxwYXRoIHN0cm9rZT0iIzMzMyIgc3Ryb2tlLWxpbmVqb2luPSJiZXZlbCIgc3Ryb2tlLW1pdGVybGltaXQ9IjEwIiBzdHJva2Utd2lkdGg9IjEuNSIgZD0iTTEgMUw5IDkuNjEwOTFMMTcgMSIvPjwvc3ZnPg==);
  background: #E5E5E5;
  color: #999999;
  cursor: not-allowed !important;
  pointer-events: none;
}

<select
  className="c0"
>
  test
</select>
`;
